### DISCUSSION TITLE
upload multiple image as binary array

### DISCUSSION THREAD

**Discussion by npltr62** [PostgREST/postgrest#2923] [Q&A] [answered]
I want upload not one image but multiple image from select input, in order to have a row with bytea array column.
```
<input type="file" id="fileInput" multiple> 

const fileInput = document.getElementById("fileInput");

const files = fileInput.files;

 const headers = {
            'Content-Profile': 'srv_tech',
            'Content-Type': 'application/octet-stream',
          };
          const response = await this.$axios.post('rpc/upload_binary', files, { headers });
```

**Comment by steve-chavez** [comment]
If you have a function with a single `bytea` unnamed parameter, that would work. See https://postgrest.org/en/stable/references/api/stored_procedures.html#functions-with-a-single-unnamed-parameter.

**Reply by npltr62** [reply]
I have a function with a single bytea[] unnamed parameter but it seems that's not recongnizing by api
```
CREATE OR REPLACE FUNCTION srv_tech.upload_b(bytea[])
 RETURNS integer
 LANGUAGE plpgsql
AS $function$
DECLARE
   id integer;
BEGIN
    INSERT INTO srv_tech.form_results(files) values ($1)
    RETURNING form_results.id
    INTO id;
    RETURN id;
END;
$function$
;
```

**Reply by steve-chavez** [reply]
Try making it non-array.

**Reply by npltr62** [reply]
> Try making it non-array.

i tried it and it was ok but with an array it's not working ...

**Reply by steve-chavez** [reply, marked as answer]
Yeah array will not work - bytea is an unknown binary input, the server cannot know how to separate different bytea values.
